lib/TokenManager.js

```diff
@@ -120,25 +129,31 @@ function refresh(sdk, tokenMgmtRef, storage, key) {
   }
 
   // Remove existing autoRefresh timeout for this key
-  clearRefreshTimeout(tokenMgmtRef, key);
-
-  return sdk.token.refresh(token)
-  .then(function(freshToken) {
-    add(sdk, tokenMgmtRef, storage, key, freshToken);
-    tokenMgmtRef.emitter.emit('refreshed', key, freshToken, token);
-    return freshToken;
-  })
-  .fail(function(err) {
-    if (err.name === 'OAuthError') {
-      remove(tokenMgmtRef, storage, key);
-      emitExpired(tokenMgmtRef, key, token);
-    }
-    throw err;
-  });
+  clearExpireEventTimeout(tokenMgmtRef, key);
```

Comment:
Since this a private function that's only called by our `getAsync`, can we also remove the preceding token check since it's already being done by getAsync? And probably just pass token down instead of re-asking storage for it.